core/wiki/macros/tree.tid

```diff
@@ -1,53 +1,64 @@
+<span>{{$:/core/images/file}}</span>
+<$macrocall $name="leaf-link" full-title=<<full-title>> chunk=<<chunk>>/>
+</$list>
 </$list>
 </li>
 \end
 
-\define branch-node(prefix)
-<$reveal type="nomatch" text="" default="""$(chunk)$""">
-<li>      
-<$list filter="""[all[shadows+tiddlers]prefix[$prefix$$(chunk)$]] -[[$prefix$$(chunk)$]] +[limit[1]]""">
-<$reveal type="nomatch" state="""$:/state/tree/$prefix$$(chunk)$""" text="show">
-<$button set="""$:/state/tree/$prefix$$(chunk)$""" setTo="show" class="tc-btn-invisible">{{$:/core/images/folder}} <$text text="""$(chunk)$"""/></$button>
+\define branch-node(prefix, chunk)
+<li>
+<$set name="reveal-state" value={{{ [[$:/state/tree/]addsuffix<__prefix__>addsuffix<__chunk__>] }}}>
+<$reveal type="nomatch" state=<<reveal-state>> text="show">
+<$button set=<<reveal-state>> setTo="show" class="tc-btn-invisible">
+{{$:/core/images/folder}} <$text text=<<__chunk__>>/>
+</$button>
 </$reveal>
-<$reveal type="match" state="""$:/state/tree/$prefix$$(chunk)$""" text="show">
-<$button set="""$:/state/tree/$prefix$$(chunk)$""" setTo="hide" class="tc-btn-invisible">{{$:/core/images/folder}} <$text text="""$(chunk)$"""/></$button>
+<$reveal type="match" state=<<reveal-state>> text="show">
+<$button set=<<reveal-state>> setTo="hide" class="tc-btn-invisible">
+{{$:/core/images/folder}} <$text text=<<__chunk__>>/>
+</$button>
 </$reveal>
-</$list>
-<$list filter="""[all[shadows+tiddlers]prefix[$prefix$$(chunk)$]] -[[$prefix$$(chunk)$]] +[limit[1]]"""><span>(<$count filter="""[all[shadows+tiddlers]prefix[$prefix$$(chunk)$]] -[[$prefix$$(chunk)$]]"""/>)</span>
-<$reveal type="match" state="""$:/state/tree/$prefix$$(chunk)$""" text="show">
-<$macrocall $name="tree-node" prefix="""$prefix$$(chunk)$"""/>
+<span>
+(<$count filter="[all[shadows+tiddlers]] +[removeprefix<__prefix__>removeprefix<__chunk__>] -[<__prefix__>addsuffix<__chunk__>]"/>)
+</span>
+<$reveal type="match" state=<<reveal-state>> text="show">
+<$macrocall $name="tree-node" prefix={{{ [<__prefix__>addsuffix<__chunk__>] }}}/>
 </$reveal>
-</$list>
+</$set>
 </li>
-</$reveal>
 \end
 
 \define tree-node(prefix)
 <ol>
-<$list filter="[all[shadows+tiddlers]removeprefix[$prefix$]splitbefore[/]sort[title]] +[!suffix[/]]" variable="chunk">
-<<leaf-node """$prefix$""">>
+<$list filter="[all[shadows+tiddlers]] +[removeprefix<__prefix__>] +[splitbefore[/]] +[sort[]] +[!suffix[/]]" variable="chunk">
+<$macrocall $name="leaf-node" prefix=<<__prefix__>> chunk=<<chunk>>/>
 </$list>
-<$list filter="[all[shadows+tiddlers]removeprefix[$prefix$]splitbefore[/]sort[title]] +[suffix[/]]" variable="chunk">
-<<branch-node """$prefix$""">>
+<$list filter="[all[shadows+tiddlers]] +[removeprefix<__prefix__>] +[splitbefore[/]] +[sort[]] +[suffix[/]]" variable="chunk">
+<$macrocall $name="branch-node" prefix=<<__prefix__>> chunk=<<chunk>>/>
 </$list>
 </ol>
 \end
 
-\define tree(prefix)
+\define tree(prefix: "$:/")
 <div class="tc-tree">
-<span><$text text="""$prefix$"""/></span>
```

Comment:
Hi @morosanuae I'm on mobile and so can't double check, but I think that the lack of whitespace inside this span was deliberate. With the newlines introduced the text nodes of the tree will gain a newline character, too. It won't be visible in the rendered DOM, but it is significant for JS code that looks directly at the DOM.

(A scenario where this matters is when generating a tree within a static HTML file with JS code for dynamically expanding/collapsing nodes).

Reply:
Hi @morosanuae of course, I forgot that you created the original.

(The background is that any text in between HTML elements/widgets is put into text nodes, and that includes whitespace, so the version with newlines is actually equivalent to `<span><$text text=<<new-line-character>>/><$text text-"""$prefix$"""/><$text text=<<new-line-character>>/></span>`. As I say, the newline characters will just appear as whitespace, not as an actual line break, and so the effect is seldom noticeable).

I would recommend changing it to the single line form because it does give neater output.

Reply:
@Jermolene ok, thanks for explaining. I'll close this one and republish it after making the changes.